utils/helper_cloud.sh

```diff
@@ -0,0 +1,697 @@
+
+  echo "Connectors: creating topics and ACLs for service account $serviceAccount"
+  ccloud kafka acl create --allow --service-account $serviceAccount --operation READ --consumer-group connect-replicator
+  ccloud kafka acl create --allow --service-account $serviceAccount --operation describe --cluster-scope
+
+  return 0
+}
+
+function ccloud_demo_preflight_check() {
+  CLOUD_KEY=$1
+  CONFIG_FILE=$2
+  enable_ksql=$3
+
+  if [ -z "$enable_ksql" ]; then
+    enable_ksql=true
+  fi
+
+  ccloud_validate_environment_set || exit 1
+  ccloud_cli_set_kafka_cluster_use "$CLOUD_KEY" "$CONFIG_FILE" || exit 1
+  validate_confluent_cloud_schema_registry "$SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO" "$SCHEMA_REGISTRY_URL" || exit 1
+  if $enable_ksql ; then
+    validate_ccloud_ksql "$KSQL_ENDPOINT" "$CONFIG_FILE" "$KSQL_BASIC_AUTH_USER_INFO" || exit 1
+  fi
+}
+
+function ccloud_validate_environment_set() {
+  ccloud environment list | grep '*' &>/dev/null || {
+    echo "ERROR: could not determine if environment is set. Run 'ccloud environment list' andset 'ccloud environment use' and try again"
+    exit 1
+  }
+
+  return 0
+
+}
+
+function ccloud_cli_set_kafka_cluster_use() {
+  CLOUD_KEY=$1
+  CONFIG_FILE=$2
+
+  if [[ "$CLOUD_KEY" == "" ]]; then
+    echo "ERROR: could not parse the broker credentials from $CONFIG_FILE. Verify your credentials and try again."
+    exit 1
+  fi
+  kafkaCluster=$(ccloud api-key list | grep "$CLOUD_KEY" | awk '{print $8;}')
+  if [[ "$kafkaCluster" == "" ]]; then
+    echo "ERROR: Could not associate key $CLOUD_KEY to a Confluent Cloud Kafka cluster. Verify your credentials, ensure the API key has a set resource type, and try again."
+    exit 1
+  fi
+  ccloud kafka cluster use $kafkaCluster
+  echo -e "\nAssociated key $CLOUD_KEY to Confluent Cloud Kafka cluster $kafkaCluster:"
+  ccloud kafka cluster describe $kafkaCluster
+
+  return 0
+}
+
+function cloud_create_demo_stack() {
+  enable_ksql=$1
+
+  RANDOM_NUM=$((1 + RANDOM % 1000000))
+  echo "RANDOM_NUM: $RANDOM_NUM"
+
+  SERVICE_NAME="demo-app-$RANDOM_NUM"
+  SERVICE_ACCOUNT_ID=$(cloud_create_service_account $SERVICE_NAME)
+  echo "SERVICE_ACCOUNT_ID: $SERVICE_ACCOUNT_ID, SERVICE_NAME: $SERVICE_NAME"
+
+  ENVIRONMENT_NAME="demo-env-$SERVICE_ACCOUNT_ID"
+  ENVIRONMENT=$(cloud_create_and_use_environment $ENVIRONMENT_NAME)
+  echo "ENVIRONMENT: $ENVIRONMENT, ENVIRONMENT_NAME: $ENVIRONMENT_NAME"
+
+  CLUSTER_NAME=demo-kafka-cluster-$SERVICE_ACCOUNT_ID
+  CLUSTER_CLOUD="${CLUSTER_CLOUD:-aws}"
+  CLUSTER_REGION="${CLUSTER_REGION:-us-west-2}"
+  CLUSTER=$(cloud_create_and_use_cluster $CLUSTER_NAME $CLUSTER_CLOUD $CLUSTER_REGION)
+  BOOTSTRAP_SERVERS=$(ccloud kafka cluster describe $CLUSTER -o json | jq -r ".endpoint" | cut -c 12-)
+  CLUSTER_CREDS=$(cloud_create_credentials_resource $SERVICE_ACCOUNT_ID $CLUSTER)
+  echo "CLUSTER: $CLUSTER, BOOTSTRAP_SERVERS: $BOOTSTRAP_SERVERS, CLUSTER_CREDS: $CLUSTER_CREDS"
```

Comment:
The output gets rather lengthy with these informational lines that have multiple values, especially the ones with keys and urls.  FYC print every value onto it's own line, that also makese copy and paste easier than having to navigate the long line and comma

Reply:
@rspurgeon these few statements are more for debugging if needed, and not useful _when all is working well_.  I actually have thought to remove them completely, especially because their contents are written to the config file.  WDYT about this latter approach

Reply:
I'm on the fence with the printed output.  I think it's good to see the progress.  I agree with the suggestion of printing every value on its own line.  But I don't have a strong opinion, so only writing to a config file seems to work as well.

Reply:
Cleaned this all up in the last commit